### PR TITLE
Fix extra backticks in smart single quotes (#1286)

### PR DESCRIPTION
**Pull request by blueglyph**
```
boolean word = FormatAsciiDocAction.isWord(editor.getDocument(), selectionStart, selectionEnd, "" + c);
if (word || c == '"' || c == '\'' || c == '~' || c == '^') {
```

Explanation: word was true for "` so it appeared to work when the selection was on word boundaries and double quotes were used, but
- word is false for '` (see the isWord method), so it was not detected correctly
- selection outside word boundaries wouldn't work with backticks at all => needs to test explicitly that the previous character is a backtick

## What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Documentation
- [ ] Other, please describe:

## Does this PR introduce a breaking change?

- [x] No
- [ ] Yes

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/asciidoctor/asciidoctor-intellij-plugin/blob/main/CONTRIBUTING.adoc) document.
- [x] I have given my PR a descriptive name. If it resolves a specific issue, I referenced it in the form of `fix #xxx` (where `xxx` is the issue number).
- [ ] I have updated the [documentation](https://github.com/asciidoctor/asciidoctor-intellij-plugin/blob/main/doc/users-guide/modules/ROOT/pages/features/) to reflect the changes introduced by this PR.
- [ ] I have added tests to cover the changes in the code.
- [ ] I need help writing a test (Maintainers are willing to help).

Creating the pull request triggers some automated tests.
See the [chapter building in the contributor's guide](https://intellij-asciidoc-plugin.ahus1.de/docs/contributors-guide/coder/building-and-running.html) to find out how to run them on your local machine.

I could build the plugin and test it manually, it was fine.
**Two unit tests didn't pass** but they don't pass with the unmodified repo either:
- AsciiDocWrapperTest
  - testShouldRenderPlantUmlAsPng
- AsciiDocCreateMissingFileInspectionTest
  - testCreateMissingIncludeFile

### CODE REVIEW

**Comment by blueglyph** [comment]
For information, the failed tests:

1) testShouldRenderPlantUmlAsPng

```
11:29:32,605 SEVERE                 #TestFramework - Test failed?
java.lang.AssertionError?
	at org.junit.Assert.fail(Assert.java:87)?
	at org.junit.Assert.assertTrue(Assert.java:42)?
	at org.junit.Assert.assertTrue(Assert.java:53)?
	at org.asciidoc.intellij.AsciiDocWrapperTest.testShouldRenderPlantUmlAsPng(AsciiDocWrapperTest.java:44)?
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)?
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)?
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)?
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)?
	at junit.framework.TestCase.runTest(TestCase.java:177)?
	at com.intellij.testFramework.UsefulTestCase.lambda$runBare$11(UsefulTestCase.java:484)?
	at com.intellij.testFramework.UsefulTestCase.lambda$wrapTestRunnable$13(UsefulTestCase.java:505)?
	at com.intellij.testFramework.UsefulTestCase.runTestRunnable(UsefulTestCase.java:409)?
	at com.intellij.testFramework.fixtures.BasePlatformTestCase.runTestRunnable(BasePlatformTestCase.java:105)?
	at com.intellij.testFramework.UsefulTestCase.defaultRunBare(UsefulTestCase.java:425)?
	at com.intellij.testFramework.UsefulTestCase.lambda$runBare$12(UsefulTestCase.java:492)?
	at com.intellij.testFramework.EdtTestUtil.lambda$runInEdtAndWait$1(EdtTestUtil.java:40)?
	at com.intellij.openapi.application.TransactionGuardImpl.runWithWritingAllowed(TransactionGuardImpl.java:209)?
	at com.intellij.openapi.application.TransactionGuardImpl.access$100(TransactionGuardImpl.java:21)?
	at com.intellij.openapi.application.TransactionGuardImpl$1.run(TransactionGuardImpl.java:191)?
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:813)?
	at com.intellij.openapi.application.impl.ApplicationImpl$3.run(ApplicationImpl.java:429)?
	at com.intellij.openapi.application.impl.LaterInvocator$1.run(LaterInvocator.java:97)?
	at com.intellij.openapi.application.impl.FlushQueue.doRun(FlushQueue.java:74)?
	at com.intellij.openapi.application.impl.FlushQueue.runNextEvent(FlushQueue.java:114)?
	at com.intellij.openapi.application.impl.FlushQueue.flushNow(FlushQueue.java:36)?
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:318)?
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:779)?
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:730)?
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:724)?
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)?
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)?
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:749)?
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:909)?
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:756)?
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$5(IdeEventQueue.java:437)?
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:774)?
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$6(IdeEventQueue.java:436)?
	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:113)?
	at com.intellij.ide.IdeEventQueue.performActivity(IdeEventQueue.java:615)?
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$7(IdeEventQueue.java:434)?
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:813)?
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:480)?
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207)?
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128)?
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:117)?
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:113)?
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:105)?
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:92)?
?

java.lang.AssertionError
	at org.junit.Assert.fail(Assert.java:87)
	at org.junit.Assert.assertTrue(Assert.java:42)
	at org.junit.Assert.assertTrue(Assert.java:53)
	at org.asciidoc.intellij.AsciiDocWrapperTest.testShouldRenderPlantUmlAsPng(AsciiDocWrapperTest.java:44)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at junit.framework.TestCase.runTest(TestCase.java:177)
	at com.intellij.testFramework.UsefulTestCase.lambda$runBare$11(UsefulTestCase.java:484)
	at com.intellij.testFramework.UsefulTestCase.lambda$wrapTestRunnable$13(UsefulTestCase.java:505)
	at com.intellij.testFramework.UsefulTestCase.runTestRunnable(UsefulTestCase.java:409)
	at com.intellij.testFramework.fixtures.BasePlatformTestCase.runTestRunnable(BasePlatformTestCase.java:105)
	at com.intellij.testFramework.UsefulTestCase.defaultRunBare(UsefulTestCase.java:425)
	at com.intellij.testFramework.UsefulTestCase.lambda$runBare$12(UsefulTestCase.java:492)
	at com.intellij.testFramework.EdtTestUtil.lambda$runInEdtAndWait$1(EdtTestUtil.java:40)
	at com.intellij.openapi.application.TransactionGuardImpl.runWithWritingAllowed(TransactionGuardImpl.java:209)
	at com.intellij.openapi.application.TransactionGuardImpl.access$100(TransactionGuardImpl.java:21)
	at com.intellij.openapi.application.TransactionGuardImpl$1.run(TransactionGuardImpl.java:191)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:813)
	at com.intellij.openapi.application.impl.ApplicationImpl$3.run(ApplicationImpl.java:429)
	at com.intellij.openapi.application.impl.LaterInvocator$1.run(LaterInvocator.java:97)
	at com.intellij.openapi.application.impl.FlushQueue.doRun(FlushQueue.java:74)
	at com.intellij.openapi.application.impl.FlushQueue.runNextEvent(FlushQueue.java:114)
	at com.intellij.openapi.application.impl.FlushQueue.flushNow(FlushQueue.java:36)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:318)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:779)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:730)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:724)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:749)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:909)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:756)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$5(IdeEventQueue.java:437)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:774)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$6(IdeEventQueue.java:436)
	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:113)
	at com.intellij.ide.IdeEventQueue.performActivity(IdeEventQueue.java:615)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$7(IdeEventQueue.java:434)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:813)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:480)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:117)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:105)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:92)
```

2) testCreateMissingIncludeFile

[...]
```
11:29:33,689 FINE   aemon.impl.PassExecutorService -  WolfHighlightingPass; id=9 Finished but there are passes in the queue: 2; progress=1378423942 V: ':var: ab?
?
...'?
11:29:33,693 FINE   openapi.util.io.FileSystemUtil - D:\src?
java.nio.file.NoSuchFileException: D:\src?
	at java.base/sun.nio.fs.WindowsException.translateToIOException(WindowsException.java:85)?
	at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:103)?
	at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:108)?
	at java.base/sun.nio.fs.WindowsFileAttributeViews$Basic.readAttributes(WindowsFileAttributeViews.java:53)?
	at java.base/sun.nio.fs.WindowsFileAttributeViews$Basic.readAttributes(WindowsFileAttributeViews.java:38)?
	at java.base/sun.nio.fs.WindowsFileSystemProvider.readAttributes(WindowsFileSystemProvider.java:199)?
	at java.base/java.nio.file.Files.readAttributes(Files.java:1851)?
	at com.intellij.openapi.util.io.NioFiles.readAttributes(NioFiles.java:177)?
	at com.intellij.openapi.util.io.FileSystemUtil$Nio2MediatorImpl.getAttributes(FileSystemUtil.java:276)?
	at com.intellij.openapi.util.io.FileSystemUtil.getAttributes(FileSystemUtil.java:88)?
	at com.intellij.openapi.vfs.impl.local.LocalFileSystemBase.getAttributesWithCustomTimestamp(LocalFileSystemBase.java:786)?
	at com.intellij.openapi.vfs.DiskQueryRelay.lambda$accessDiskWithCheckCanceled$0(DiskQueryRelay.java:52)?
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)?
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)?
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)?
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:702)?
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:699)?
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)?
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:699)?
	at java.base/java.lang.Thread.run(Thread.java:833)?

[...]

11:29:33,881 SEVERE                 #TestFramework - test failed: Comparison Failure?
Actual [[:var: ab?
?
<error descr="include file not found: D:/src/ab/missing.adoc">include::{var}/missing.adoc[]</error>?
]]?
Expected [[:var: ab?
?
<error descr="include file not found: C:/src/ab/missing.adoc">include::{var}/missing.adoc[]</error>?
]]?
?
com.intellij.rt.execution.junit.FileComparisonFailure: createMissingIncludeFile.adoc: extra (3:1/29): 'include::{var}/missing.adoc[]' (include file not found: D:/src/ab/missing.adoc) [HighlightInfoTypeImpl[severity=ERROR, key=ERRORS_ATTRIBUTES]]?
createMissingIncludeFile.adoc: missing (3:1/29): 'include::{var}/missing.adoc[]' (include file not found: C:/src/ab/missing.adoc)?
 expected:<...ude file not found: [C]:/src/ab/missing.ado...> but was:<...ude file not found: [D]:/src/ab/missing.ado...>?
	at com.intellij.testFramework.ExpectedHighlightingData.compareTexts(ExpectedHighlightingData.java:624)?
	at com.intellij.testFramework.ExpectedHighlightingData.checkResult(ExpectedHighlightingData.java:575)?
	at com.intellij.testFramework.ExpectedHighlightingData.checkResult(ExpectedHighlightingData.java:511)?
	at com.intellij.testFramework.fixtures.impl.CodeInsightTestFixtureImpl.collectAndCheckHighlighting(CodeInsightTestFixtureImpl.java:1671)?
	at com.intellij.testFramework.fixtures.impl.CodeInsightTestFixtureImpl.collectAndCheckHighlighting(CodeInsightTestFixtureImpl.java:1634)?
	at com.intellij.testFramework.fixtures.impl.CodeInsightTestFixtureImpl.collectAndCheckHighlighting(CodeInsightTestFixtureImpl.java:1621)?
	at com.intellij.testFramework.fixtures.impl.CodeInsightTestFixtureImpl.testHighlighting(CodeInsightTestFixtureImpl.java:538)?
	at com.intellij.testFramework.fixtures.impl.CodeInsightTestFixtureImpl.testHighlighting(CodeInsightTestFixtureImpl.java:594)?
	at org.asciidoc.intellij.inspections.AsciiDocQuickFixTestBase.configure(AsciiDocQuickFixTestBase.java:39)?
	at org.asciidoc.intellij.inspections.AsciiDocQuickFixTestBase.doTest(AsciiDocQuickFixTestBase.java:22)?
	at org.asciidoc.intellij.inspections.AsciiDocCreateMissingFileInspectionTest.testCreateMissingIncludeFile(AsciiDocCreateMissingFileInspectionTest.java:17)?
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)?
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)?
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)?
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)?
	at junit.framework.TestCase.runTest(TestCase.java:177)?
	at com.intellij.testFramework.UsefulTestCase.lambda$runBare$11(UsefulTestCase.java:484)?
	at com.intellij.testFramework.UsefulTestCase.lambda$wrapTestRunnable$13(UsefulTestCase.java:505)?
	at com.intellij.testFramework.UsefulTestCase.runTestRunnable(UsefulTestCase.java:409)?
	at com.intellij.testFramework.fixtures.BasePlatformTestCase.runTestRunnable(BasePlatformTestCase.java:105)?
	at com.intellij.testFramework.UsefulTestCase.defaultRunBare(UsefulTestCase.java:425)?
	at com.intellij.testFramework.UsefulTestCase.lambda$runBare$12(UsefulTestCase.java:492)?
	at com.intellij.testFramework.EdtTestUtil.lambda$runInEdtAndWait$1(EdtTestUtil.java:40)?
	at com.intellij.openapi.application.TransactionGuardImpl.runWithWritingAllowed(TransactionGuardImpl.java:209)?
	at com.intellij.openapi.application.TransactionGuardImpl.access$100(TransactionGuardImpl.java:21)?
	at com.intellij.openapi.application.TransactionGuardImpl$1.run(TransactionGuardImpl.java:191)?
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:813)?
	at com.intellij.openapi.application.impl.ApplicationImpl$3.run(ApplicationImpl.java:429)?
	at com.intellij.openapi.application.impl.LaterInvocator$1.run(LaterInvocator.java:97)?
	at com.intellij.openapi.application.impl.FlushQueue.doRun(FlushQueue.java:74)?
	at com.intellij.openapi.application.impl.FlushQueue.runNextEvent(FlushQueue.java:114)?
	at com.intellij.openapi.application.impl.FlushQueue.flushNow(FlushQueue.java:36)?
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:318)?
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:779)?
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:730)?
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:724)?
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)?
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)?
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:749)?
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:909)?
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:756)?
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$5(IdeEventQueue.java:437)?
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:774)?
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$6(IdeEventQueue.java:436)?
	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:113)?
	at com.intellij.ide.IdeEventQueue.performActivity(IdeEventQueue.java:615)?
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$7(IdeEventQueue.java:434)?
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:813)?
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:480)?
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207)?
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128)?
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:117)?
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:113)?
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:105)?
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:92)?
?

[...]
```

**Comment by blueglyph** [comment]
And yes, I have no idea how to add a regression test for this particular case. 😅

**Comment by ahus1** [comment]
Hi @blueglyph - thanks for preparing this PR. 

I had a look at the change, and sometimes two backticks are necessary. Assume the following, where always `test` is selected: 

```
test -> `test`
xtestx -> x``test``x
test's -> ``test``'s
'test' -> '`test`'
```

You see that I also updated a test in the updated PR. 

For the local tests, don't worry to much; one is a local Windows configuration issue. I should make the test more resilient, but that's outside of this PR.
The other one might be a problem with a local PlantUML configuration. I'd leave that for another day/PR.

As you've checked out the project, do you want to give it a test run with `gradlew -Dfile.encoding=UTF-8 runIde`?

**Comment by blueglyph** [comment]
Thanks for reviewing and modifying it, I didn't know the double backtick requirement.

I've tested it, it does work as expected with the list you gave above of course, and a few more cases with single and double quotes.

That's probably not a real problem but it fails when there are spaces on the left and/or the right. That's why I modified it as I did (ignoring two backticks were necessary inside a word).

If I select `.test.`, or `.test` or `test.` (where `.` = space, so we can see them), then type quotes and backticks, it ends up with quotes and double backticks, like `'``.test``'.` in the first example.

After a quick test, this seems to address it, unless I miss other cases:

```diff
--- a/src/main/java/org/asciidoc/intellij/actions/FormattingQuotedTypedHandler.java
+++ b/src/main/java/org/asciidoc/intellij/actions/FormattingQuotedTypedHandler.java
@@ -56,10 +56,16 @@ public class FormattingQuotedTypedHandler extends TypedHandlerDelegate {
           newText = selectedText;
         }
         // ... or add it around the text
+        boolean quoted = false;
+        if (selectionStart > 0 && selectionEnd < editor.getDocument().getTextLength()) {
+          String surroundings = editor.getDocument().getText(TextRange.create(selectionStart - 1, selectionEnd + 1));
+          quoted = (surroundings.startsWith("'") && surroundings.endsWith("'"))
+            || (surroundings.startsWith("\"") && surroundings.endsWith("\""));
+        }
         int border = 0;
         if (newText == null) {
           boolean word = FormatAsciiDocAction.isWord(editor.getDocument(), selectionStart, selectionEnd, String.valueOf(c));
-          if (word || c == '"' || c == '\'' || c == '~' || c == '^') {
+          if (word || c == '"' || c == '\'' || c == '~' || c == '^' || (c == '`' && quoted)) {
             newText = c + selectedText + c;
             border = 1;
           } else {
```

But... I suppose it's not worth going that far, I don't see why one would want to do that anyway. What do you think? Too convoluted? Or should I commit it?

**Comment by ahus1** [comment]
Thanks for trying this. Still I see that the typographic quotes have some corner cases when you add a blank before the test. 

A text like the following (with dots representing blanks as in your previous comment)

```
'`.test.`'
```

This renders only the second quote will be typographic, while the first won't be rendered as a typographic quote - see the following screen shot: 

image

So I assume that for the cases where there is a blank at the start, it is OK to used double backticks as only monospace is possible. 

This leaves the blank at the end ... while this works, I find this very uncommon. So I suggest to leave it as is. 

So if you're ok with the way it is now, I'll squash this PR and prepare a pre-release. 

BTW: I fixed the problems in the unit tests in the main branch; the tests should now pass on windows regardless of the drive. The other test case with PlantUML should (after a change) now either pass, or show a better error message.

**Comment by ahus1** [comment]
Thank you for your participation! I have to admit I do a lot trial and error with the preview to find corner cases.
